Automattic/jetpack
# Super Cache: fix how the plugin uses apache_request_headers

If a site has [disabled the function](https://wordpress.org/support/topic/module-is-not-working-if-apache_request_headers-is-disabled-2/) `apache_request_headers()` then WPSC will not display the "Rejected User Agents" textarea on the settings page. It would also cause a fatal error because the code checked if the function existed, before creating a fallback of the same name.

## Proposed changes:

* Create a function wpsc_apache_request_headers() that is a fallback for that function, and it checks if the original function exists and is callable before calling it. Otherwise, it extracts the headers.
* Remove the apache_request_headers() checks that were added 6 years ago to hide the Rejected User Agent settings.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





* If you're using the Jetpack Docker container, login to it with `jetpack docker sh`
* Edit the php.ini of your server. If the above container, it'll be `/etc/php/8.2/fpm/php.ini`
* Look for "disable_functions =" and add `apache_request_headers` at the end.
* Restart PHP with `service php8.2-fpm restart`
* Go to the WPSC settings, and enable "304 Browser caching".
* Enable debugging in the plugin and watch the logs in a new browser tab.
* Open a new _private_ window and load the front page of your site.
* You should see a log entry: "304 browser caching not possible as timestamps differ"
* Reload the page, and you may see this: "Send 304 Not Modified header."
* You could also error_log $remote_mod_time in wp-cache-phase2.php on line 289 and line 293 to make sure they match.
* Don't forget to reverse changes to php.ini and restart PHP.
